enzyme/test/Enzyme/ReverseMode/mpi_rust.ll

```diff
@@ -0,0 +1,115 @@
+; RUN: if [ %llvmver -eq 15 ]; then %opt < %s %loadEnzyme -enzyme -opaque-pointers=1 -S | FileCheck %s; fi
+; RUN: if [ %llvmver -ge 15 ]; then %opt < %s %newLoadEnzyme -passes="enzyme" -opaque-pointers=1 -S | FileCheck %s; fi
+
+; ModuleID = 'enzyme-repro.ll'
+source_filename = "dot_enzyme.3df87ea89a38df43-cgu.0"
+target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+@RSMPI_DOUBLE = external local_unnamed_addr global ptr
+@RSMPI_COMM_WORLD = external local_unnamed_addr global ptr
+@RSMPI_COMM_SELF = external local_unnamed_addr global ptr
+
+; Function Attrs: noinline nonlazybind sanitize_hwaddress uwtable
+define hidden noundef "enzyme_type"="{[-1]:Float@double}" double @_ZN10dot_enzyme12dot_parallel17h7dfcd86d9e8c176bE(ptr noalias nocapture noundef readonly align 8 dereferenceable(16) "enzyme_type"="{[-1]:Pointer}" %0, ptr noalias nocapture noundef nonnull readonly align 8 "enzyme_type"="{[-1]:Pointer, [-1,-1]:Float@double}" %1, i64 noundef "enzyme_type"="{[-1]:Integer}" %2, ptr noalias nocapture noundef nonnull readonly align 8 "enzyme_type"="{[-1]:Pointer, [-1,-1]:Float@double}" %3, i64 noundef "enzyme_type"="{[-1]:Integer}" %4, ptr noundef "enzyme_type"="{[0]:Pointer}" %5) unnamed_addr #1 personality ptr @rust_eh_personality {
+  %7 = alloca double, align 8
+  %8 = alloca double, align 8
+  call void @llvm.lifetime.start.p0(i64 8, ptr nonnull %8)
+  %9 = alloca double, align 8
+  store double 1.000, ptr %8, align 8
+  call void @llvm.lifetime.start.p0(i64 8, ptr nonnull %7)
+  store double 0.000000e+00, ptr %7, align 8
+  tail call void @llvm.experimental.noalias.scope.decl(metadata !7)
+  %10 = load ptr, ptr @RSMPI_DOUBLE, align 8, !noalias !10, !noundef !13
```

Comment:
This is unused atm so this won't test any thing. Can you make this a minimal runnable case. Maybe a different mpi fn?

Also get rid of the other stuff like enzyme_type etc

Reply:
@jedbrown I can't run things, can you test if some other function works on this branch?